monai/data/dataset.py

```diff
@@ -810,12 +811,11 @@ def _compute_cache():
             # only compute cache for the unique items of dataset
             mapping = {self.hash_func(v): v for v in data}
             self.data = list(mapping.values())
-            cache_ = _compute_cache()
-            self._cache = dict(zip(list(mapping)[: self.cache_num], cache_))
-            self.data = data
+            self._hash_keys = list(mapping)[: self.cache_num]
```

Comment:
self.cache_num seems to be always 0 at this point, so your self._hash_keys will be always empty. !?

Reply:
Thanks for your finding, yes, I made a mistake here.
I updated in the latest commit and added a new unit test to cover it.

Thanks.